### PR TITLE
made changes to ping driver

### PR DESCRIPTION
**Pull request by aryanbattula**
refactored ping driver

### CODE REVIEW

**Review by ianwood103 — DISMISSED** [review]
looks good to me but also didn't test it

File: EosPayload/drivers/ping_driver.py [hunk 1]
```diff
@@ -55,63 +53,47 @@ def ping_reply(self, client, user_data, message):
...
         header = DataHeader(
-            data_type=Type.TELEMETRY,
+            data_type=Type.PING,
             sender=self.get_device_id(),
-            priority=Priority.TELEMETRY
+            priority=Priority.TELEMETRY,
         )
-        packet = Packet(bytes(command, 'utf8'), header)
+
+        packet = Packet(Ping(PingEnum.PING, counter), header)
```
Comment (VanL15) [inline EosPayload/drivers/ping_driver.py:95]: I think the PingEnum should be ACK and not PING

**Review by VanL15 — CHANGES_REQUESTED** [review]

**Review by IonLi09 — DISMISSED** [review]
lgtm

File: EosPayload/drivers/ping_driver.py [hunk 2]
```diff
@@ -27,17 +31,11 @@
...
-        PING = "PING"
-        ACK = "ACK"
-        ERR = "ERR"
-
     def setup(self) -> None:
         super().setup()
 
         # TODO: remove this when ping command format is implemented
-        raise Exception("Ping driver is temporary disabled until ping format class is implemented")
+        # raise Exception("Ping driver is temporary disabled until ping format class is implemented")
```
Comment (VanL15) [inline EosPayload/drivers/ping_driver.py:38]: remove this and the comment above it

File: EosPayload/drivers/ping_driver.py [hunk 3]
```diff
@@ -55,63 +53,47 @@ def ping_reply(self, client, user_data, message):
...
-            data_type=Type.TELEMETRY,
+            data_type=Type.PING,
             sender=self.get_device_id(),
-            priority=Priority.TELEMETRY
+            priority=Priority.TELEMETRY,
         )
-        packet = Packet(bytes(command, 'utf8'), header)
+
+        packet = Packet(Ping(PingEnum.PING, counter), header)
+        # packet = Packet(Ping(True, counter), header)
```
Comment (VanL15) [inline EosPayload/drivers/ping_driver.py:96]: remove this line

File: EosPayload/drivers/radio_driver.py [hunk 4]
```diff
@@ -89,12 +89,13 @@ def data_receive_callback(xbee_message):
             packet = xbee_message.data  # raw bytearray packet
             logger.info("Packet received ~~~~~~")
             logger.info(packet)
-            packet_object = Packet.decode(packet)  # convert packet bytearray to packet object
+            packet_object = Packet.decode(bytes(packet))  # convert packet bytearray to packet object
+            data_header = packet_object.data_header
```
Comment (VanL15) [inline EosPayload/drivers/radio_driver.py:93]: remove this line, `data_header` isn't used

File: EosPayload/drivers/radio_driver.py [hunk 5]
```diff
@@ -89,12 +89,13 @@ def data_receive_callback(xbee_message):
...
 
             # Try to data log the packet, but we really don't want to block in a callback
             if self.log_lock.acquire(blocking=False):
                 try:
-                    self.data_log(["received", packet_object.encode_to_string()])
+                    self.data_log(["received"])
```
Comment (VanL15) [inline EosPayload/drivers/radio_driver.py:98]: change this to like "Received packet". Add this above it `# TODO change this to include packet as a string (encode_to_string function) `

File: EosPayload/drivers/radio_driver.py [hunk 6]
```diff
@@ -104,7 +105,9 @@ def data_receive_callback(xbee_message):
             dest = packet_object.data_header.destination  # packet object
             if dest in self.device_map:  # mapping from device to mqtt topic
                 mqtt_topic = self.device_map[dest]
-                self._mqtt.send(mqtt_topic, packet)
+                # if isinstance(packet, bytearray):
+                #     packet = bytes(packet)
```
Comment (VanL15) [inline EosPayload/drivers/radio_driver.py:109]: remove this comment and the once above it

File: EosPayload/drivers/radio_driver.py [hunk 7]
```diff
@@ -121,7 +124,7 @@ def xbee_send_callback(_client, _userdata, message):
                 # Store to data file
                 if self.log_lock.acquire(blocking=False):
                     try:
-                        self.data_log(["sent", packet_from_mqtt.encode_to_string()])
+                        self.data_log(["sent"])
```
Comment (VanL15) [inline EosPayload/drivers/radio_driver.py:127]: Change this to "Sent packet". Also add this above it `# TODO change this to include packet as a string (encode_to_string function) `

File: scripts/mqtt_pub_packet.py [hunk 8]
```diff
@@ -28,7 +29,8 @@
 
     if not isinstance(args.body, str):
         raise Exception("body must be a str")
-    body = bytes(args.body, 'utf8')
+    # body = bytes(args.body, 'utf8')
```
Comment (VanL15) [inline scripts/mqtt_pub_packet.py:32]: remove this line

**Review by VanL15 — CHANGES_REQUESTED** [review]
see comments

**Review by VanL15 — APPROVED** [review]
merge :)

**Pull request merged**